airflow/models/taskinstance.py

```diff
@@ -848,11 +848,15 @@ def refresh_from_db(self, session: Session = NEW_SESSION, lock_for_update: bool
         """
         self.log.debug("Refreshing TaskInstance %s from DB", self)
 
-        qry = session.query(TaskInstance).filter(
-            TaskInstance.dag_id == self.dag_id,
-            TaskInstance.task_id == self.task_id,
-            TaskInstance.run_id == self.run_id,
-            TaskInstance.map_index == self.map_index,
+        qry = (
+            session.query(TaskInstance)
+            .filter(
+                TaskInstance.dag_id == self.dag_id,
+                TaskInstance.task_id == self.task_id,
+                TaskInstance.run_id == self.run_id,
+                TaskInstance.map_index == self.map_index,
+            )
```

Comment:
I guess it would be good to leave a comment why we are doing it

Reply:
I wonder if there is a way of saying to SQLA "only select this, never select anything else" (i.e. to prevent other regressions in the future

Reply:
I guess (from the discussion) this "feature" of SQL of locking the other side is kinda undocumented and dangerous ... and something unexpected as welll (breakes the "explicit is better than implicit" zen of Python) so I am afraid we might not find a "nicer" way of disabling it, because it feels like a hack.

Reply:
Maybe we should add some automation checking it - It should be ony when you select Taskinstance with lock, which should be kinda possible to do with some clever AST analysis.

Reply:
I've left a comment, and the best way I found to do it was to do `query(TI.__table__.columns)` which will only ever select the specified columns and never any relationships -- that way (with the comment) I don't think the pre-commit is needed